server/paymentProviders/stripe/virtual-cards.js

```diff
@@ -247,6 +263,37 @@ export const processTransaction = async (stripeTransaction, stripeEvent) => {
   });
 };
 
+export const processUpdatedTransaction = async (stripeAuthorization, stripeEvent) => {
+  const virtualCard = await getVirtualCardForTransaction(stripeAuthorization.card);
+  if (!virtualCard) {
+    logger.error(`Stripe: could not find virtual card ${stripeAuthorization.card.id}`, stripeEvent);
+    return;
+  }
+
+  if (stripeEvent) {
+    await checkStripeEvent(virtualCard.host, stripeEvent);
+  }
+
+  if (stripeAuthorization.status === 'reversed') {
```

Comment:
Could it be that the expense was already reversed and canceled on our side? We would then do this multiple time.

Reply:
I think it is worth adding a check here and updating only if necessary. Thanks!